### PR TITLE
{tools}[foss/2019b] BOINC v7.16.5

### PR DESCRIPTION
**Pull request by tardigradus**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+name = 'BOINC'
+version = '7.16.5'
+versionsuffix = "-client"
+
+homepage = 'https://boinc.berkeley.edu'
+description = """BOINC is a program that lets you donate your idle computer time to science projects
+ like SETI@home, Climateprediction.net, Rosetta@home, World Community Grid, and many others."""
+
+toolchain = {'name': 'foss', 'version': '2019b'}
```
Comment (Micket) [inline easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb]: GCCcore would surely suffice for this?

File: easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        'tag': 'client_release/%(version_major_minor)s/%(version)s',
+    },
+}]
+checksums = ['d8c808682d0549b7e1045f1b4633aeb8287619014effc4c16c7b4145b089f988']
+
+dependencies = [
+]
+
+builddependencies = [
+]
```
Comment (Micket) [inline easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb]: Why remove all the builddeps? Definitely looks like at least some of them are needed.

File: easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    'git_config': {
+        'url': 'https://github.com/BOINC',
+        'repo_name': 'boinc',
+        'tag': 'client_release/%(version_major_minor)s/%(version)s',
+    },
+}]
+checksums = ['d8c808682d0549b7e1045f1b4633aeb8287619014effc4c16c7b4145b089f988']
+
+dependencies = [
+]
```
Comment (Micket) [inline easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb]: Old version had cURL as a dependency; doesn't that still apply?

We should also list openssl as os dep
```
osdependencies = [('openssl-devel', 'libssl-dev', 'libopenssl-devel')]
```

File: easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb (hunk 1 above)
Comment (tardigradus) [reply easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb]: Yes, I'll change this.

File: easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb (hunk 2 above)
Comment (tardigradus) [reply easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb]: How do I work out what is needed?  Should, say, 'make' be specified even though the easyblock is ConfigureMake?

File: easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb (hunk 3 above)
Comment (tardigradus) [reply easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb]: What's the policy regarding whether something should be considered a dependency?  The binaries created are linked against a whole bunch of libraries, including curl, but also stuff like
libldap-2.4.so.2 and libbz2.so.1.

File: easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb (hunk 2 above)
Comment (Micket) [reply easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb]: Autotools based projects should normally provide a pre-generated configure and Makefile (which leads to the classic `./configure; make; make install` procedure). But, for generating the scripts to start (which it looks like `./_autosetup` is doing), you need all the Autotools stuff.

For `make` itself, i must admit I'm not sure.

File: easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb (hunk 3 above)
Comment (Micket) [reply easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb]: Some exceptions when we want the OS to provide security updates. OpenSSL falls into this category. I asked the other maintainers, and they though we can add openldap into that category as well. So, please add a `('openldap-devel', 'libldap2-dev')` (or possible disable ldap if it is optional and not important for cluster use, I'm not sure if it is). I don't know what the package names are in ubuntu, `s

For bzip2, cURL , there are packages, and we should definitely link against those.

File: easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb (hunk 2 above)
Comment (tardigradus) [reply easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb:27]: Added Autotools.

Comment (Micket) [reply easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb:27]: Just in case you missed it ; you haven't actually updated the PR

Comment (tardigradus) [reply easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb:36]: Doesn't pushing to the PR branch update it? Or are you referring to the fact that I resolved the conversation before actually pushing the commit?

Comment (Micket) [reply easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb:36]: It does, it was just that i didn't see any new changes even an hour after the comment, so i thought you might have missed it.

File: easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+builddependencies = [
+    ('Autotools', '20180311'),
+]
+
+dependencies = [
+    ('cURL', '7.66.0'),
+]
+
+osdependencies = [
+    ('openssl-devel', 'libssl-dev', 'libcurl-devel'),
```
Comment (Micket) [inline easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb:35]: Typo? cURL is a anormal dependency. libopenssl-devel is the alternative ssl package name on some distros 
```suggestion
    ('openssl-devel', 'libssl-dev', 'libopenssl-devel'),
```

Comment (tardigradus) [reply easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb:35]: There is a dependence on `libcurl-devel`.  Without it, `configure` throws the following error:

    ERROR: could not find (recent enough) development-libs for libcurl

Isn't it right to include this as an OS dependency?

Regarding the name of SSL devel package, doesn't EB do some magic to find the correct package name for the platform?

Comment (Micket) [reply easybuild/easyconfigs/b/BOINC/BOINC-7.16.5-foss-2019b-client.eb:35]: 1. It should (we should make it) find the libcurl headers from the cURL easyconfig dependency.

2. The only "magic" is that you specify a tuple of alternative names, openssl-devel, libssl-dev, libopenssl-devel are the alterantive names. libcurl-devel does not belong in the same tuple. If we also had a libcurl os dependency (which I don't think we should) we should specify a seperate tuple with it's alternative names, like
```
osdependencies = [
    ('libcurl-devel', 'libcurl-dev'),
    ('openssl-devel', 'libssl-dev', 'libopenssl-devel'),
]
```
Though again; it's already a normal dependency, it shouldn't also be a osdependency.

**Comment by akesandgren** [comment]
Toolchains is old, and this doesn't really fit the normal target for software in EasyBuild.
So closing.
If you want to bring this up again make a new PR with a more recent toolchain.

**Pull request closed without merging**